src/components/Nav/Nav.js

```diff
@@ -1,7 +1,602 @@
 import React from 'react';
+import styled from 'styled-components';
+import { Link, useNavigate } from 'react-router-dom';
+import { useState, useEffect, useRef } from 'react';
 
 const Nav = () => {
-  return <div>Nav</div>;
+  const [bottomMenuDisplay, setBottomMenuDisplay] = useState('');
+  const [inputValue, setInputValue] = useState('');
+
+  const navigate = useNavigate();
+
+  // Event : 헤더 검색어 받아오기
+  const inputValueRecord = event => {
+    setInputValue(event.target.value);
+  };
+
+  // Event : 검색어 입력 후 페이지 이동 및 검색어 전달
+  const inputPageMoveHandler = () => {
+    navigate('/search', {
+      state: {
+        inputValue: inputValue,
+      },
+    });
+
+    setInputValue('');
+  };
```

Comment:
* 객체의 key,value가 같은경우 ES6문법인 `단축속성명`을 적용 할 수 있습니다.
```suggestion
  const inputPageMoveHandler = () => {
    navigate('/search', {
      state: {
        inputValue
      },
    });

    setInputValue('');
  };
```